File: src/app/cuisine.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { Cuisine } from './cuisine';
import { CuisineDetails } from './cuisineDetails';

@Injectable({
  providedIn: 'root'
})
export class CuisineService {

  baseUrl:string="http://localhost:8088/cuisines"
  constructor(private http:HttpClient) { }

  getAllCuisines():Observable<Cuisine[]>{
    
    return this.http.get<Cuisine[]>(this.baseUrl+"/getAllCuisines");
  }

  deleteCuisine(cuisineId:any):Observable<any>{
    alert("Inside deleteCuisine in Cuisine Service");
    alert(this.baseUrl+"/deleteCuisine/"+cuisineId)
    return this.http.delete(this.baseUrl+"/deleteCuisine/"+cuisineId,{ responseType: 'text' });
  }

  deleteCuisineItem(cuisine_details_id:any):Observable<any>{
    alert("Inside deleteCuisineItem in Cuisine Service");
    alert(this.baseUrl+"/deleteCuisineItem/"+cuisine_details_id)
    return this.http.delete(this.baseUrl+"/deleteCuisineItem/"+cuisine_details_id,{ responseType: 'text' });
  }

  getCuisineItems(cuisineId:any):Observable<any>{

    return this.http.get<CuisineDetails[]>(this.baseUrl+"/getCuisineItems/"+cuisineId);

  }
  updateCuisineItem(cuisine_details_id:any, cuisineItem:any):Observable<CuisineDetails>{
    
    return this.http.put<CuisineDetails>(this.baseUrl+"/updateCuisineItem/"+cuisine_details_id,cuisineItem); 

  }
  saveCuisineItem(cuisineItem:any):Observable<CuisineDetails>{

    return this.http.post<CuisineDetails>(this.baseUrl,cuisineItem);
  }
  getCuisineItemById(cuisine_details_id:any):Observable<any>{
    return this.http.get(this.baseUrl+"/getCuisineItemById/"+cuisine_details_id);
  }

  addCuisineItem(cuisineId:any,cuisineItemsInfo:any):Observable<CuisineDetails>{
    return this.http.post<CuisineDetails>(this.baseUrl+"/addCuisineItem/"+cuisineId,cuisineItemsInfo)
  }
}

File: src/main/java/com/simplilearn/fooddelivery/service/CuisineService.java
package com.simplilearn.fooddelivery.service;

import java.util.List;
import java.util.Optional;

import com.simplilearn.fooddelivery.dto.CuisineDetailsDto;
import com.simplilearn.fooddelivery.dto.CuisineDto;
import com.simplilearn.fooddelivery.entity.Cart;
import com.simplilearn.fooddelivery.entity.Cuisine_Details;


public interface CuisineService {
	
	public List<CuisineDto> getAllCuisines();
	public boolean deleteCuisine(int cuisineId);
	public boolean deleteCuisineItem(int cuisine_details_id);

	public List<CuisineDetailsDto> getCuisineItems(int cuisineId);
	public void updateCuisineItem(Cuisine_Details cuisineItem,int cuisine_details_id);
	public boolean addCuisineItem(Cuisine_Details cd,int cuisineId);
	
	public Optional<Cuisine_Details> getCuisineItemById(int cuisine_details_id);

	
}
